sdk/JavaScript/TestFramework/js/tests/pushTests.js

```diff
@@ -76,298 +77,187 @@ function definePushTestsNamespace() {
-                    test.addLog('success: Registered template', templateName);
-                    pushChannel = channel;
-                    complete(true);
-                }, function (error) {
-                    test.addLog('Error regsitering template push channel: ', error);
-                    complete(false);
-                });
-            }, function (error) {
-                test.addLog('Error creating push channel: ', error);
-                complete(false);
-            });
-        });
-    }
-
-    function UnRegisterTemplateForPush(templateName, test) {
-        return new WinJS.Promise(function (complete) {
-            if (pushChannel) {
-                zumo.getClient().push.unregisterTemplate(templateName).done(function () {
-                    test.addLog('Template Unregistered with NH: ', templateName);
-                    complete(true);
-                }, function (error) {
-                    test.addLog('Failed to unregister template with NH: ', error);
-                    complete(false);
-                });
-            } else {
-                test.addLog('Error, push channel needs to be registered.');
-                complete(false);
-            }
-            pushChannel = null;
-        });
-    }
-
-    var indexOfTemplatePushTests = 0;
+    };
 
-    tests.push(new zumo.Test('Register Template toast push channel', function (test, done) {
-        var templateBody = '<toast><visual><binding template="ToastText01"><text id="1">$(News_French)</text></binding></visual></toast>';
-        var wnsHeaders = { 'X-WNS-Type': 'wns/toast' };
-        var tags = ["World", "French"];
-        RegisterTemplateForPush(templateBody, toastTemplateName, wnsHeaders, tags, test).done(function (result) {
-            done(result);
-        });
-    }));
-    tests.push(createPushTest('sendToastText01',
-        { text1: 'This is not used in template tests' },
-        '<toast><visual><binding template="ToastText01"><text id="1">Nouvelles du monde en français!</text></binding></visual></toast>', true));
-    tests.push(new zumo.Test('Unregister Toast Template push channel', function (test, done) {
-        UnRegisterTemplateForPush(toastTemplateName, test).done(function (result) {
-            done(result);
-        });
+    tests.push(new zumo.Test('InitialDeleteRegistrations', function (test, done) {
+        testPlatform.getPushChannel()
+            .then(function (channel) {
+                channelUri = channel.uri;
+                return zumo.getClient().push.unregisterAll(channel.uri);
+            })
+            .then(getRegistrations(channelUri, 'wns'))
+            .then(function (registrations) {
+                zumo.assert.isTrue(Array.isArray(registrations));
+                zumo.assert.areEqual(0, registrations.length);
+                zumo.assert.areEqual('{}', JSON.stringify(zumo.getClient().push._registrationManager._storageManager._registrations));
```

Comment:
That's a good point.  I'm using the internal methods to see what registrations exist and I think that is a valid use.  Another option would be to do what is done with v2 tests & call an api to verify the registrations on that end.

I think you're right that asserting that the storageManager contains the valid registration info is in excess, I copied a lot of the registration verification logic from unit tests.  It would look quite a bit cleaner without it.